### PR TITLE
GHA-integration initial commit

### PR DESCRIPTION
**Pull request by Agarwal-Shivansh**
I have commented revive check to pass the lint check, we have to enable it in the later phase

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        run: go build github.com/onosproject/onos-control/pkg/...
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - uses: golangci/golangci-lint-action@v3.7.0
```
Comment (gab-arrobo) [inline .github/workflows/master.yml]:
```suggestion
      - uses: golangci/golangci-lint-action@v4.0.0
```

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          go-version-file: 'go.mod'
+      - name: Unit tests
+        run: go test -race github.com/onosproject/onos-control/pkg/...
+
+  license-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: reuse lint
+        uses: fsfe/reuse-action@v2
```
Comment (gab-arrobo) [inline .github/workflows/master.yml]:
```suggestion
        uses: fsfe/reuse-action@v2
```

**Review by gab-arrobo — APPROVED** [review]
+1

**Comment by gab-arrobo** [comment]
retest this please

**Comment by Agarwal-Shivansh** [comment]
> retest this please

Retested it.

**Pull request merged**